src/OSPSuite.Core/Domain/ObjectPath.cs

```diff
@@ -146,6 +149,15 @@ public virtual T Resolve<T>(IEntity refEntity) where T : class
          return resolvePath<T>(dependentObject, usePath);
       }
 
+      private void removeMatchingParentPath(IContainer root, List<string> usePath)
+      {
+         var matchingElements = !string.IsNullOrEmpty(root?.ParentPath?.PathAsString) &&
+                                new ObjectPath(_pathEntries).PathAsString.StartsWith(root?.ParentPath?.PathAsString ?? string.Empty);
```

Comment:
that ?? can never happen since you check before

Reply:
the code is kind of weird. we are using _pathEntries (global) to modify the parameter pass (usePath). 
this works because usePath is created using _pathEntries but this is really hard to follow. 
Also we have lists, we got to string, we compare strings and we then remove from list.
Should't we just use...no string at all?